Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -255,13 +246,75 @@ protected RandomCutForest(Builder<?> builder) {
         if (parallelExecutionEnabled) {
             // If the user specified a thread pool size, use it. Otherwise, use available
             // processors - 1.
-            threadPoolSize = builder.threadPoolSize.orElse(Runtime.getRuntime().availableProcessors() - 1);
             traversalExecutor = new ParallelForestTraversalExecutor(components, threadPoolSize);
-            updateExecutor = new ParallelForestUpdateExecutor(updateCoordinator, components, threadPoolSize);
+            updateExecutor = new ParallelForestUpdateExecutor(this.updateCoordinator, components, threadPoolSize);
         } else {
-            threadPoolSize = 0;
             traversalExecutor = new SequentialForestTraversalExecutor(components);
-            updateExecutor = new SequentialForestUpdateExecutor(updateCoordinator, components);
+            updateExecutor = new SequentialForestUpdateExecutor(this.updateCoordinator, components);
+        }
+    }
+
+    private void initCompact() {
+        PointStoreDouble pointStore = new PointStoreDouble(dimensions, sampleSize * numberOfTrees + 1);
```

Comment:
question. why +1 capacity?

Reply:
I think this is to handle the case when the point store is completely full and we need to update the forest with a new point. The new point has to be added to the point store. If it's accepted into a tree, then at least one other point will be removed from the point store. If it's not accepted by any tree, than the submitted point will be removed from the point store at the end of the update operation.